src/util/stream_generator.h

```diff
@@ -46,93 +33,124 @@ class suspend_sometimes {
   constexpr void await_resume() const noexcept {}
 };
 
-/**
- * The promise type that backs the generator type and handles storage and
- * suspension-related decisions.
- */
-template <size_t MIN_BUFFER_SIZE>
+// The promise type that backs the generator type and handles storage and
+// suspension-related decisions.
+template <size_t BUFFER_SIZE>
 class stream_generator_promise {
-  std::ostringstream _stream;
-  std::exception_ptr _exception;
+  std::array<char, BUFFER_SIZE> data_;
+  size_t currentIndex_ = 0;
+  static_assert(BUFFER_SIZE > 0, "Buffer size must be greater than zero");
+  // Temporarily store data that didn't fit into the buffer so far.
+  std::string_view overflow_;
+  std::exception_ptr exception_;
 
  public:
-  using value_type = std::stringbuf;
+  using value_type = std::string_view;
   using reference_type = value_type&;
```

Comment:
reference_type can also be `string_view`, and `value_type` should probably be `std::string`, otherwise you have strange semantics in principle.